From 7ad1703640896451fa6563fc331122bf201b7a60 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mathieu=20Dupr=C3=A9?= <mathieu.dupre@savoirfairelinux.com>
Date: Thu, 11 May 2023 11:13:26 +0200
Subject: [PATCH] grub-efi: remove unneeded and untested items
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Remove files and instructions that were not tested when
GRUB_SIGN_VERIFY is set to 0 and SE_LOADER is set to 1.

Signed-off-by: Sébastien LE STUM <sebastien.le-stum@savoirfairelinux.com>
Signed-off-by: Mathieu Dupré <mathieu.dupre@savoirfairelinux.com>
---
 .../grub/grub-efi-efi-secure-boot.inc         | 30 +++----------------
 1 file changed, 4 insertions(+), 26 deletions(-)

diff --git a/sources/meta-secure-core/meta-efi-secure-boot/recipes-bsp/grub/grub-efi-efi-secure-boot.inc b/sources/meta-secure-core/meta-efi-secure-boot/recipes-bsp/grub/grub-efi-efi-secure-boot.inc
index 5814461..8fbb75c 100644
--- a/sources/meta-secure-core/meta-efi-secure-boot/recipes-bsp/grub/grub-efi-efi-secure-boot.inc
+++ b/sources/meta-secure-core/meta-efi-secure-boot/recipes-bsp/grub/grub-efi-efi-secure-boot.inc
@@ -127,14 +127,6 @@ do_install:append:class-target() {
     # Install the stacked grub configs.
     install -d "${D}${EFI_BOOT_PATH}"
     install -m 0600 "${WORKDIR}/grub-efi.cfg" "${D}${EFI_BOOT_PATH}/grub.cfg"
-    install -m 0600 "$menu" "${D}${EFI_BOOT_PATH}"
-    [ x"${UEFI_SB}" = x"1" ] && {
-        install -m 0600 "${WORKDIR}/efi-secure-boot.inc" "${D}${EFI_BOOT_PATH}"
-        install -m 0600 "${WORKDIR}/password.inc" "${D}${EFI_BOOT_PATH}"
-    }
-
-    # Create the initial environment block with empty item.
-    grub-editenv "${D}${EFI_BOOT_PATH}/grubenv" create
 
     install -d "${D}${EFI_BOOT_PATH}/${GRUB_TARGET}-efi"
     grub-mkimage -c ../cfg -p "${GRUB_PREFIX_DIR}" -d "./grub-core" \
@@ -185,11 +177,6 @@ addtask sign after do_install before do_deploy do_package
 
 fakeroot do_chownboot() {
     chown root:root -R "${D}${EFI_BOOT_PATH}/grub.cfg${SB_FILE_EXT}"
-    chown root:root -R "${D}${EFI_BOOT_PATH}/boot-menu.inc${SB_FILE_EXT}"
-    [ x"${UEFI_SB}" = x"1" ] && {
-        chown root:root -R "${D}${EFI_BOOT_PATH}/efi-secure-boot.inc${SB_FILE_EXT}"
-        chown root:root -R "${D}${EFI_BOOT_PATH}/password.inc${SB_FILE_EXT}"
-    }
 }
 addtask chownboot after do_deploy before do_package
 
@@ -197,18 +184,12 @@ addtask chownboot after do_deploy before do_package
 do_deploy:append:class-target() {
     install -m 0644 "${D}${EFI_BOOT_PATH}/${GRUB_IMAGE}" "${DEPLOYDIR}"
 
+    if [ "${GRUB_IMAGE}" != "${GRUB_IMAGE_PREFIX}${GRUB_IMAGE}" ]; then
+        install -m 0664 "${DEPLOYDIR}/${GRUB_IMAGE}" "${DEPLOYDIR}/${GRUB_IMAGE_PREFIX}${GRUB_IMAGE}"
+    fi
+
     # Deploy the stacked grub configs.
-    install -m 0600 "${D}${EFI_BOOT_PATH}/grubenv" "${DEPLOYDIR}"
     install -m 0600 "${D}${EFI_BOOT_PATH}/grub.cfg" "${DEPLOYDIR}"
-    install -m 0600 "${D}${EFI_BOOT_PATH}/boot-menu.inc" "${DEPLOYDIR}"
-    install -m 0600 "${D}${EFI_BOOT_PATH}/grub.cfg${SB_FILE_EXT}" "${DEPLOYDIR}"
-    install -m 0600 "${D}${EFI_BOOT_PATH}/boot-menu.inc${SB_FILE_EXT}" "${DEPLOYDIR}"
-    [ x"${UEFI_SB}" = x"1" ] && {
-        install -m 0600 "${D}${EFI_BOOT_PATH}/efi-secure-boot.inc" "${DEPLOYDIR}"
-        install -m 0600 "${D}${EFI_BOOT_PATH}/password.inc" "${DEPLOYDIR}"
-        install -m 0600 "${D}${EFI_BOOT_PATH}/efi-secure-boot.inc${SB_FILE_EXT}" "${DEPLOYDIR}"
-        install -m 0600 "${D}${EFI_BOOT_PATH}/password.inc${SB_FILE_EXT}" "${DEPLOYDIR}"
-    }
 
     install -d "${DEPLOYDIR}/efi-unsigned"
     install -m 0644 "${B}/${GRUB_IMAGE}" "${DEPLOYDIR}/efi-unsigned"
@@ -219,7 +200,4 @@ FILES:${PN} += "${EFI_BOOT_PATH}"
 
 CONFFILES:${PN} += "\
     ${EFI_BOOT_PATH}/grub.cfg \
-    ${EFI_BOOT_PATH}/grubenv \
-    ${EFI_BOOT_PATH}/boot-menu.inc \
-    ${EFI_BOOT_PATH}/efi-secure-boot.inc \
 "
-- 
2.25.1

